tests/backend/mongodb/test_indexes.py

```diff
@@ -0,0 +1,11 @@
+import pytest
+
+pytestmark = pytest.mark.bdb
+
+
+def test_asset_id_index():
```

Comment:
I don't really like this. The main reason is that we are not actually testing the queries that we have.

All the points you mentioned in https://github.com/bigchaindb/bigchaindb/pull/1131#issue-204602113 are valid.
Maybe the Lazy object that @vrde is working on will help or maybe we need to find a different to solution to test the secondary indexes.

I propose that we postpone secondary index tests and handle them correctly in #1126